### ISSUE TITLE
vTHL4: Level A compliancy with the Directive on the accessibility

### ISSUE THREAD

**Issue by jaakkocsc** [CSCfi/rems#1172]
Currently, official requirements are only stated in Finnish as: "Saavutettavuusvaatimuksilla tarkoitetaan tässä käytännössä WCAG 2.1 -ohjeistuksen A- ja AA-tason kriteerejä."

**Comment by opqdonut** [comment]
Is this only for applicant UIs? Handlers also? Administration?

**Comment by luontola** [comment]
https://www.w3.org/WAI/WCAG21/quickref/ has a checklist of things that are needed for different compliancy levels. Here is a non-comprehensive list of things that will probably require changes:

1.4.3 Contrast (Minimum) - Level AA
1.4.11 Non-text Contrast -Level AA
- need to check the contrast levels

1.4.10 Reflow - Level AA
- our tables are too wide for mobile and require horizontal scrolling

2.1.1 Keyboard - Level A
- the "Show more" links in the event log are not focusable with Tab; need to check others too

2.4.1 Bypass Blocks - Level A
- we need a "jump to content" link

2.4.2 Page Titled - Level A
- the pages need titles

2.4.7 Focus Visible - Level AA
- focus for the links to change language is not very visible (it's only underlined)
- secondary buttons have a very low contrast focus color

3.2.1 On Focus - Level A
3.2.5 Change on Request - Level AAA
- links which open in a new window need a warning about it (currently they only have an icon; aria stuff might be needed) https://www.w3.org/TR/WCAG20-TECHS/H83.html https://www.w3.org/TR/WCAG20-TECHS/G200.html https://www.w3.org/TR/WCAG20-TECHS/G201.html

3.3.1 Error Identification - Level A
- error reporting needs work to be accessible to screen readers

4.1.2 Name, Role, Value - Level A
- not all headers are currently HTML headings, for example application form sections
- some pages use `<h2>` instead of `<h1>` as the top level heading

Also the navigation doesn't currently move focus to the page content after navigating to a page using keyboard. The way it should work is demonstrated here: https://youtu.be/3tgz1E4MsAk?t=309

**Comment by luontola** [comment]
Here is another good resource about web accessibility: https://a11yproject.com/

**Comment by luontola** [comment]
For accessibility testing it's possible to consult Annanpura: https://www.annanpura.fi/

**Comment by luontola** [comment]
http://papunet.net/saavutettavuus/saavutettavien-verkkosivujen-suunnitteluopas

**Comment by opqdonut** [comment]
Need to decide how to track A11Y status. Scheduled? Per release?
